### ISSUE TITLE
az storage blob delete-batch --dryrun deleted blobs

### ISSUE THREAD

**Issue by sq-code** [Azure/azure-cli#17237]
## Describe the bug

**Command Name**
`az storage blob delete-batch`

**Errors:**
Command didn't generate a dry run log, it went ahead and deleted. Big problem, since the --pattern flag didn't work as I expected it to either (probably my error, verifying the pattern match is why I wanted to see a dry run) 

## To Reproduce:
I have not tried to reproduce, but basically my scenario was the following: 

1. have general purpose v2 storage account
2. create a container
3. upload some pseudo-hierarchical blobs, something like:
    container/folderA/blob1
    container/folderA/blob2
    container/folderB/blob1
    container/folderB/blob2
4. run the following command, substituting authentication pieces as appropriate:
$CONTAINER_NAME="container"
$blob="folderA"
az storage blob delete-batch \
--source $CONTAINER_NAME \
--account-key $ACCOUNT_KEY \
--account-name $STORAGE_ACCOUNT_NAME \
--auth-mode key \
--connection-string $CONNECTION_STRING \
--delete-snapshots include \
--dryrun \
--pattern $blob\
--subscription $SUBSCRIPTION
5. give it a bit and refresh the portal. In my case, everything in the container was deleted. I assume the pattern I used matched  all the blobs, although that was not what I intended and I've not bothered to try to work out if that is actually what happened. If the pattern does match limited blobs, probably only the blobs it matches will be deleted.

## Expected Behavior
I expect a report when I include the --dryrun flag, or an error if I'm running a version that doesn't recognize --dryrun as a valid flag. I do not expect any deletions to occur regardless of what other flags, patterns, etc are part of the command.

## Environment Summary
```
macOS-11.2.2-x86_64-i386-64bit
Python 3.8.8
Installer: HOMEBREW

azure-cli 2.20.0

```

**Comment by yonzhan** [comment]
storage

**Comment by sq-code** [comment]
Went back and attempted to recreate with a test container as I described. Can't recreate, it works correctly. I did turn on soft-delete on blobs in the meantime. Only other change is that originally I was populating --pattern $blob with lines read in from a file though the loop never got beyond the first iteration. No idea what's going on and I can't recreate the original command since that data is gone.
